extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/KafkaStreamsProducer.java

```diff
@@ -0,0 +1,309 @@
+import org.apache.kafka.clients.admin.ListTopicsResult;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.streams.KafkaClientSupplier;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.KafkaStreams.StateListener;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.Topology;
+import org.apache.kafka.streams.processor.StateRestoreListener;
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.Unremovable;
+import io.quarkus.runtime.Startup;
+
+/**
+ * Manages the lifecycle of a Kafka Streams pipeline. If there's a producer
+ * method returning a KS {@link Topology}, then this topology will be configured
+ * and started. Optionally, before starting the pipeline, this manager will wait
+ * for a given set of topics to be created, as KS itself will fail without all
+ * input topics being created upfront.
+ */
+@Singleton
+public class KafkaStreamsProducer {
+
+    private static final Logger LOGGER = Logger.getLogger(KafkaStreamsProducer.class.getName());
+
+    private final ExecutorService executorService;
+    private final KafkaStreams kafkaStreams;
+    private final KafkaStreamsTopologyManager kafkaStreamsTopologyManager;
+
+    @Inject
+    public KafkaStreamsProducer(KafkaStreamsSupport kafkaStreamsSupport, KafkaStreamsRuntimeConfig runtimeConfig,
+            Instance<Topology> topology, Instance<KafkaClientSupplier> kafkaClientSupplier,
+            Instance<StateListener> stateListener, Instance<StateRestoreListener> globalStateRestoreListener) {
+        // No producer for Topology -> nothing to do
+        if (topology.isUnsatisfied()) {
+            LOGGER.debug("No Topology producer; Kafka Streams will not be started");
+            this.executorService = null;
+            this.kafkaStreams = null;
+            this.kafkaStreamsTopologyManager = null;
+            return;
+        }
+
+        Properties buildTimeProperties = kafkaStreamsSupport.getProperties();
+
+        String bootstrapServersConfig = asString(runtimeConfig.bootstrapServers);
+        Properties kafkaStreamsProperties = getStreamsProperties(buildTimeProperties, bootstrapServersConfig, runtimeConfig);
+        Properties adminClientConfig = getAdminClientConfig(kafkaStreamsProperties);
+
+        this.executorService = Executors.newSingleThreadExecutor();
+
+        this.kafkaStreams = initializeKafkaStreams(kafkaStreamsProperties, runtimeConfig, adminClientConfig, topology.get(),
+                kafkaClientSupplier, stateListener, globalStateRestoreListener, executorService);
+        this.kafkaStreamsTopologyManager = new KafkaStreamsTopologyManager(adminClientConfig);
+    }
+
+    @Produces
+    @Singleton
+    @Unremovable
+    @Startup
+    public KafkaStreams getKafkaStreams() {
+        return kafkaStreams;
+    }
+
+    @Produces
+    @Singleton
+    @Unremovable
```

Comment:
Same here

Reply:
It was before so I mimic the existing behavior.